File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0

    lst = [0]

    for target in moves:
        for j in range(len(board)):
            if board[j][target-1] != 0:
                lst.append(board[j][target-1])
                board[j][target-1] = 0

                tmp1 = lst.pop()
                tmp2 = lst.pop()
                if tmp1 == tmp2:
                    answer += 2
                else:
                    lst.append(tmp2)
                    lst.append(tmp1)

                break
            elif board[j][target-1] == 0:
                continue

    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <bits/stdc++.h>

using namespace std;

// [[0,0,0,0,0],[0,0,1,0,3],[0,2,5,0,1],[4,2,4,4,2],[3,5,1,3,1]]	
// [1,5,3,5,1,2,1,4]

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    
    stack<int> bascket;
    
    for(int target : moves){
        for(int i = 0; i < board.size(); i++){
            if(!board[i][target - 1] == 0){
                if(!bascket.empty() && bascket.top() == board[i][target - 1]){
                    answer += 2;
                    bascket.pop();
                }
                else{
                    bascket.push(board[i][target - 1]);
                }
                board[i][target - 1] = 0;
                break;
            }
        }
    }
    

    return answer;
}